app/assets/javascripts/browse-columns.js

```diff
@@ -303,5 +305,16 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     }
   }
 
+  BrowseColumns.prototype.showBanner = function (slug) {
```

Comment:
I'm confused why this needs JavaScript at all, but perhaps I've missed something so happy to discuss.

The logic to show/hide the intervention component is already present in the backend, and the JS gets its data from that same place (via JSON in an AJAX call, which includes a chunk of markup for the column to be rendered, which includes/doesn't include the intervention component as appropriate).

As an experiment, I checked out this branch locally, removed this JS and modified `_links.html.erb` as follows:

```
  <% if !@hide_recruitment_banner %>
    <%= render "govuk_publishing_components/components/intervention", {
      suggestion_text: "Help improve GOV.UK",
      suggestion_link_text: "Take part in user research",
      suggestion_link_url: "https://gdsuserresearch.optimalworkshop.com/treejack/35gap0ij",
      new_tab: true,
    } %>
  <% end %>
```

...and it worked as expected, showing the intervention banner only where it was required (and persisting through the JS cache that's part of the browse JS also as required).

(might need a bit of refactoring in this PR to change `if !@hide_recruitment_banner` to something clearer like `if @show_recruitment_banner`)

Reply:
Yup trying this out now. Will refactor the name in a bit.

Reply:
Done